torch/quantization/fx/_equalize.py

```diff
@@ -460,6 +520,21 @@ def convert_eq_obs(
             assert(isinstance(weight_eq_obs, _WeightEqualizationObserver))
             assert(isinstance(node.target, str))
 
+            if (modules[node.target] == weight_eq_obs):
```

Comment:
How does this if statement guarantee a functional layer?

Reply:
This is a little hacky 😅  but for functional layers, the weight equalization observer is stored as a node in the graph (ex. map entry: `(linear1_w_equalization_process, _WeightEqualizationObserver())`, `modules[linear1_w_equalization_process.target] = _WeightEqualizationObserver`). So the name that gets stored in the `weight_eq_obs_dict` is the actual weight equalization observer node's name. 

However for nn.Linear layers, the weight equalization observer is created in `update_obs_for_equalization` by checking to see if the linear layer is in the `equalization_qconfig_map` and then creating the observer based on the types specified in the dictionary. Since the node isn't actually created, the key for the map entry is the linear layer's key (ex.  map entry: `(linear, _WeightEqualizationObserver)`, `modules[linear.target] = torch.nn.functional.linear`).

So, the check is seeing if the `modules[node.target]` is equal to the weight equalization observer because `modules[node.target]` should only be equal to the observer in the functional case. Alternatively, I could store (operation node, weight eq observer) entries inside of `weight_eq_obs_dict` and then check for if the operation node is a `call_module` or `call_function` here.

Reply:
I see, thanks for the explanation. I think just by looking at this code it is not clear to me why this would mean a `call_function`. I think your suggestion of storing the node type in the `weight_eq_obs_dict` makes sense to me.